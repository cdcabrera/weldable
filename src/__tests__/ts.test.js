const path = require('path');
const ts = require('../ts');
const { contextPath } = require('../global');

describe('Typescript', () => {
  afterEach(() => {
    removeFixture('tsconfig.json');
  });

  it('should return specific properties', () => {
    expect(ts).toMatchSnapshot('specific properties');
  });

  it('should create a basic tsconfig', () => {
    const mockConsoleMessageMethod = jest.fn();
    const mockConsoleMessage = {
      success: mockConsoleMessageMethod
    };

    const results = ts.createTsConfig(
      { _BUILD_DIST_DIR: path.join(contextPath, 'dist') },
      { loader: 'ts', contextPath: tempFixturePath, isCreateTsConfig: true, isRegenTsConfig: true },
      { consoleMessage: mockConsoleMessage }
    );

    expect(results).toMatchSnapshot('basic');
    expect(mockConsoleMessageMethod).toHaveBeenCalledTimes(1);
  });

  it('should disable tsconfig messaging', () => {
    const mockConsoleMessageMethod = jest.fn();
    const mockConsoleMessage = {
      info: mockConsoleMessageMethod,
      warn: mockConsoleMessageMethod,
      success: mockConsoleMessageMethod
    };

    ts.createTsConfig(
      { _BUILD_DIST_DIR: path.join(contextPath, 'dist') },
      { loader: 'ts', contextPath: tempFixturePath, isCreateTsConfig: true, isRegenTsConfig: true },
      { isMessaging: false, consoleMessage: mockConsoleMessage }
    );

    expect(mockConsoleMessageMethod).toHaveBeenCalledTimes(0);
  });

  it('should ignore creating a config on condition', () => {
    const mockConsoleMessageMethod = jest.fn();
    const mockConsoleMessage = {
      info: mockConsoleMessageMethod
    };

    ts.createTsConfig(
      { _BUILD_DIST_DIR: path.join(contextPath, 'dist') },
      { loader: 'ts', contextPath: tempFixturePath, isCreateTsConfig: false },
      { consoleMessage: mockConsoleMessage }
    );

    expect(mockConsoleMessageMethod.mock.calls).toMatchSnapshot('ignore');
  });

  it('should create a tsconfig with a base template', () => {
    const results = ts.createTsConfig(
      { _BUILD_DIST_DIR: path.join(contextPath, 'dist') },
      {
        baseTsConfig: 'create-react-app',
        loader: 'ts',
        contextPath: tempFixturePath,
        isCreateTsConfig: true,
        isRegenTsConfig: true
      }
    );

    expect(results).toMatchSnapshot('base template');
  });

  it('should update a tsconfig with a subsequent merge', () => {
    const _BUILD_DIST_DIR = path.join(contextPath, 'dist');
    const baseResults = ts.createTsConfig(
      { _BUILD_DIST_DIR },
      {
        baseTsConfig: 'create-react-app',
        loader: 'ts',
        contextPath: tempFixturePath,
        isCreateTsConfig: true,
        isRegenTsConfig: true
      }
    );

    const mergeResults = ts.createTsConfig(
      { _BUILD_DIST_DIR },
      {
        baseTsConfig: 'strictest',
        loader: 'ts',
        contextPath: tempFixturePath,
        isCreateTsConfig: true,
        isMergeTsConfig: true
      }
    );

    expect({
      original: baseResults,
      merged: mergeResults
    }).toMatchSnapshot('merged template');
  });
});
