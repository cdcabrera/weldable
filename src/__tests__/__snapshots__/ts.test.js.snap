// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Typescript should create a basic tsconfig: basic 1`] = `
{
  "compilerOptions": {
    "allowJs": true,
    "noEmit": false,
    "outDir": "dist",
  },
}
`;

exports[`Typescript should create a tsconfig with a base template: base template 1`] = `
{
  "compilerOptions": {
    "allowJs": true,
    "allowSyntheticDefaultImports": true,
    "esModuleInterop": true,
    "isolatedModules": true,
    "jsx": "react-jsx",
    "lib": [
      "dom",
      "dom.iterable",
      "esnext",
    ],
    "module": "esnext",
    "moduleResolution": "bundler",
    "noEmit": true,
    "noFallthroughCasesInSwitch": true,
    "outDir": "dist",
    "resolveJsonModule": true,
    "skipLibCheck": true,
    "strict": true,
    "target": "es2015",
  },
}
`;

exports[`Typescript should ignore creating a config on condition: ignore 1`] = `
[
  [
    "Ignoring tsconfig.json",
  ],
]
`;

exports[`Typescript should return specific properties: specific properties 1`] = `
{
  "createTsConfig": [Function],
}
`;

exports[`Typescript should update a tsconfig with a subsequent merge: merged template 1`] = `
{
  "merged": {
    "compilerOptions": {
      "allowJs": true,
      "allowSyntheticDefaultImports": true,
      "allowUnreachableCode": false,
      "allowUnusedLabels": false,
      "checkJs": true,
      "esModuleInterop": true,
      "exactOptionalPropertyTypes": true,
      "isolatedModules": true,
      "jsx": "react-jsx",
      "lib": [
        "dom",
        "dom.iterable",
        "esnext",
      ],
      "module": "esnext",
      "moduleResolution": "bundler",
      "noEmit": true,
      "noFallthroughCasesInSwitch": true,
      "noImplicitOverride": true,
      "noImplicitReturns": true,
      "noPropertyAccessFromIndexSignature": true,
      "noUncheckedIndexedAccess": true,
      "noUnusedLocals": true,
      "noUnusedParameters": true,
      "outDir": "dist",
      "resolveJsonModule": true,
      "skipLibCheck": true,
      "strict": true,
      "target": "es2015",
    },
  },
  "original": {
    "compilerOptions": {
      "allowJs": true,
      "allowSyntheticDefaultImports": true,
      "esModuleInterop": true,
      "isolatedModules": true,
      "jsx": "react-jsx",
      "lib": [
        "dom",
        "dom.iterable",
        "esnext",
      ],
      "module": "esnext",
      "moduleResolution": "bundler",
      "noEmit": true,
      "noFallthroughCasesInSwitch": true,
      "outDir": "dist",
      "resolveJsonModule": true,
      "skipLibCheck": true,
      "strict": true,
      "target": "es2015",
    },
  },
}
`;
